core/src/main/java/org/infinispan/protostream/LazyByteArrayOutputStream.java

```diff
@@ -0,0 +1,27 @@
+package org.infinispan.protostream;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+
+public interface LazyByteArrayOutputStream {
```

Comment:
nitpicking but can we get a new name? I have no idea what lazy means in this context. 

My not-so-great suggestion is `ProtoOutputStream`

Javadoc would be nice since this is a public API.

Reply:
I've gone with `RawByteArrayOutputStream` and `RawByteArrayOutputStreamImpl` for now.